### ISSUE TITLE
`make rust` is failing because `default_ref()` is missing from `raft-proto` structs

### ISSUE THREAD

**Issue by kennytm** [pingcap/kvproto#419]
Repro steps:

1. Clone the repository, everything clean
2. `make rust`
3. Got the error at the end:

```
error[E0599]: no function or associated item named `default_ref` found for type `raft_proto::prost::eraftpb::Entry` in the current scope
   --> src/prost/wrapper_debugpb.rs:348:44
    |
348 |             None => super::eraftpb::Entry::default_ref(),
    |                                            ^^^^^^^^^^^ function or associated item not found in `raft_proto::prost::eraftpb::Entry`

error[E0599]: no function or associated item named `default_ref` found for type `raft_proto::prost::eraftpb::Message` in the current scope
   --> src/prost/wrapper_raft_serverpb.rs:103:46
    |
103 |             None => super::eraftpb::Message::default_ref(),
    |                                              ^^^^^^^^^^^ function or associated item not found in `raft_proto::prost::eraftpb::Message`

error[E0599]: no function or associated item named `default_ref` found for type `raft_proto::prost::eraftpb::HardState` in the current scope
    --> src/prost/wrapper_raft_serverpb.rs:1234:48
     |
1234 |             None => super::eraftpb::HardState::default_ref(),
     |                                                ^^^^^^^^^^^ function or associated item not found in `raft_proto::prost::eraftpb::HardState`

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0599`.
error: Could not compile `kvproto`.
```

cc @nrc. I think the Rust code in both raft-proto and kvproto need to be regenerated using the same (exact) version of `protobuf-build`

**Comment by ice1000** [comment]
Try `cargo update -p protobuf-build`

**Comment by ice1000** [comment]
Try `cargo update -p protobuf-build`

**Comment by ice1000** [comment]
> Try cargo update -p protobuf-build

This isn't tikv/protobuf-build#17.

**Comment by kennytm** [comment]
@ice1000 This has nothing to do with `cargo update` since (1) this repository does not contain a `Cargo.lock`, so we will always get the latest `protobuf-build 0.6` version (i.e. 0.6.5) already, and (2) the error arises because raft-proto's [checked-in source code](https://github.com/pingcap/raft-rs/blob/master/proto/src/prost/wrapper_eraftpb.rs) is generated using 0.6.1 or earlier. `cargo update` is not going to help unless raft-proto's source is regenerated using 0.6.5.

**Comment by ice1000** [comment]
Yes -- that was why I self-replied "This isn't [the issue I thought it was duplicate of]"

**Comment by nrc** [comment]
Blocked on https://github.com/pingcap/raft-rs/pull/258

**Comment by Hoverbear** [comment]
@nrc #258 is merged

**Comment by nrc** [comment]
This should be fixed by https://github.com/pingcap/kvproto/pull/418

**Comment by Hoverbear** [comment]
@kennytm Can you confirm it's fixed?

**Comment by kennytm** [comment]
Yes it's fixed on master, thanks :)
